devito/passes/iet/misc.py

```diff
@@ -138,6 +138,8 @@ def relax_incr_dimensions(iet, options=None, **kwargs):
 def generate_macros(iet):
     applications = FindApplications().visit(iet)
     headers = set().union(*[_generate_macros(i) for i in applications])
+    # Sort for deterministic code generation
+    headers = sorted(headers)
```

Comment:
same `filter_sorted`

Reply:
For this I think filter-sorted is redundant. What would it offer more?

Reply:
I think this is conceptually wrong and useless:

Conceptually wrong because if all passes appending headers had to ensure uniquess it would be painful. BTW it's not even just the headers, it would be everything that can get passed to `metadata`.

Which brings us to the second point, useless: see here:
https://github.com/devitocodes/devito/blob/master/devito/passes/iet/engine.py#L123

Reply:
obviously if you instead added it because it generates bad code, then there's a bug somewhere else

Reply:
Correct! Thanks, it was a side-effect of the compute regions non-determinism, catching the first (min/max) area to be computed